client/src/templates/Challenges/classic/lower-jaw.tsx

```diff
@@ -178,97 +96,70 @@ const LowerJaw = ({
-      currentAttempts &&
-      testsLength &&
-      (currentAttempts >= testsLength || currentAttempts >= 3);
-
-    return (
-      <div>
-        <hr />
-        <div className='lower-jaw-icon-bar'>
-          <button
-            className='btn fade-in'
-            title={t('buttons.reset-code')}
-            aria-label={t('buttons.reset-code')}
-            data-cy='reset-code-button'
-            onClick={openResetModal}
-          >
-            <Reset />
-          </button>
-
-          {isAttemptsLargerThanTest && !challengeIsCompleted ? (
-            <button
-              className='btn fade-in'
-              id='get-help-button'
-              title={t('buttons.get-help')}
-              aria-label={t('buttons.get-help')}
-              data-cy='get-help-button'
-              onClick={openHelpModal}
-            >
-              <Help />
-            </button>
-          ) : null}
-        </div>
-      </div>
-    );
-  };
+  const isAttemptsLargerThanTest =
+    currentAttempts &&
+    testsLength &&
+    (currentAttempts >= testsLength || currentAttempts >= 3);
 
   const showDesktopButton = window.innerWidth > MAX_MOBILE_WIDTH;
 
-  const renderButtons = () => {
-    return (
-      <>
-        <div id='action-buttons-container'>
-          {isSignedIn ? null : challengeIsCompleted ? (
-            <Button
-              block={true}
-              href={`${apiLocation}/signin`}
-              className='btn-cta'
-            >
-              {t('learn.sign-in-save')}
-            </Button>
-          ) : null}
-          <button
-            id='test-button'
-            data-cy='run-tests-button'
-            className={`btn-block btn ${challengeIsCompleted ? 'sr-only' : ''}`}
-            aria-hidden={testBtnAriaHidden}
-            onClick={tryToExecuteChallenge}
-          >
-            {showDesktopButton
-              ? t('buttons.check-code')
-              : t('buttons.check-code-2')}
-          </button>
-          <button
-            id='submit-button'
-            data-cy='submit-button'
-            aria-hidden={!challengeIsCompleted}
-            className='btn-block btn'
-            onClick={tryToSubmitChallenge}
-            ref={submitButtonRef}
-          >
-            {t('buttons.submit-and-go')}
-          </button>
-        </div>
-      </>
-    );
-  };
+  const checkButton = showDesktopButton
+    ? t('buttons.check-code')
+    : t('buttons.check-code-2');
+
+  const showScreenReadSubmit = challengeIsCompleted && isEditorInFocus;
 
   return (
     <div className='action-row-container'>
-      {renderButtons()}
+      <LowerJawButtons
+        signed={isSignedIn}
+        challengeIsCompleted={challengeIsCompleted}
+        buttonText={
+          challengeIsCompleted ? t('buttons.submit-and-go') : checkButton
+        }
+        onClick={
+          challengeIsCompleted ? tryToSubmitChallenge : tryToExecuteChallenge
+        }
+        signInText={t('learn.sign-in-save')}
+      />
```

Comment:
This is a strange pattern - it makes it unclear what the text belongs to or what the click listeners are for.

Essentially what we have here is a component that contains some buttons.  Why not just define the three types of buttons as components and then conditionally render them here?

Reply:
I have done so, to preserve the focus state on the button.

Here is its test

```ts
  it('focuses the submit button after testing a valid solution', () => {
    cy.visit(location);
    cy.focused().type('{end}{enter}<meta charset="UTF-8" />');
    cy.get(selectors.lowerJawButton).should('not.be.focused');
    cy.get(selectors.lowerJawButton).click();
    cy.get(selectors.lowerJawButton).should('be.focused');
  });
  ```
  
 I think the intended behavior is for the camper to focus on the button, run the "test the solution" function, and be able to press space or enter again to start the submit function, without the need to navigate to the button with keyboard.

Reply:
Should I turn this into variables for readability? or create two buttons and use `useRef` to set the focus?

Reply:
Right, but in production the screenreader announces that the submit button exists when you click on "Check Your Code". That doesn't happen in this PR, presumably because the text is modified instead of the button as a whole being replaced.

Since this is supposed to be a refactor, it should not change that behaviour. So, yes, I think two buttons + useRef is the way to go.

Reply:
It worked without `useRef`, I still need to learn more about how the button are parsed/rendered

Reply:
It keeps focus now, which is good, but it's still not the same as production.

In production, when you click the "Check Your Code" button it announces that there's a "Submit and go to the next challenge" button on the page. That's no longer the case in this PR.

Reply:
I have tried to record my narrator reading the buttons, but couldn't because the extension doesn't catch system sounds, even after allowing it to.

I can't replicate the issue of it not reading "Submit and go to the next challenge"

Reply:
I think you should find a way to browse the site with a screen-reader since it's important to not accidentally damage it's accessibility.  One option would be to use Orca inside WSL.  That should give a better dev experience than Windows given how little we support it.

Regardless, the issue still exists, so please revert the changes to `setTestBtnAriaHidden`.

Reply:
can i suggest better title? may it be "split lower-jaw component to subcomponents"?

Reply:
> Regardless, the issue still exists, so please revert the changes to `setTestBtnAriaHidden`.

If the issue is persisting, we need to add keys to the component, so it renders a different state, then add the logic to focus on it when the challenge is complete, and make sure that it doesn't focus on when someone press `Ctrl+enter` while there aren't in the lower jaw

Reply:
> If the issue is persisting, we need to add keys to the component, so it renders a different state, then add the logic to focus on it when the challenge is complete, and make sure that it doesn't focus on when someone press `Ctrl+enter` while there aren't in the lower jaw.

I just noticed that this can be misunderstood, I have implemented the change, if it's an issue I can do more.